File: Strategy Pattern/DuckSim_java/src/main/Ente/Mechaente.java
package main.Ente;

import main.IFlugverhalten.IFlugverhalten;
import main.IQuakverhalten.IQuakverhalten;

public class Mechaente extends Ente
{
    public Mechaente(IFlugverhalten fv, IQuakverhalten qv) throws Throwable
    {
        super(fv, qv);
    }

    @Override
    public void anzeigen()
    {
        super.anzeigen();
        System.out.println("Mechaente");
    }
}

File: Strategy Pattern/DuckSim_cpp/Mechaente.cpp
#include "Mechaente.h"

Mechaente::Mechaente(IQuakverhalten* qv, IFlugverhalten* fv)
	:Ente(qv, fv)
{
}

void Mechaente::anzeigen()
{
	Ente::anzeigen();
	std::cout << "Mechaente\n";
}
